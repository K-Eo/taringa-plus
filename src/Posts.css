.post-item {
  display: flex;
  height: 18px;
  margin-bottom: 14px;
  padding-left: 10px;
}

.post-item a,
.post-item a:visited {
  display: block;
  color: #333;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  margin-left: 8px;
}

.post-icon {
  height: 18px;
  width: 18px;
  background: transparent url(./assets/images/icons.png) no-repeat scroll left top;
}

.post-item:last-child {
  margin-bottom: 0;
}

.post-item .glyphicon {
  margin-right: 10px;
}

.post-item:hover {
  color: #333;
}
