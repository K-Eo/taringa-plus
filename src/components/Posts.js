import React from 'react'
import { Card, Loader, Message } from 'semantic-ui-react'

import Post from './Post'

export const Posts = (props) => {
  const { posts, fetchControl } = props

  if (fetchControl.status === 'fetching') {
    return (
      <Loader active inline='centered'/>
    )
  }

  if (fetchControl.status === 'failure') {
    const errorMessage = `Intenta recargar la página. Error: ${fetchControl.error}`

    return (
      <Message
        icon='frown'
        header='Houston, tenemos un problema!'
        content={errorMessage}
        negative
      />
    )
  }

  if (fetchControl.status === 'success' && posts.length === 0) {
    return (
      <Message
        icon='smile'
        header='No hay posts!'
        content='Prueba recargando la página para intentar descargar los últimos posts.'
      />
    )
  }

  return (
    <Card.Group itemsPerRow={4}>
      {
        posts.map(post => <Post key={post.id} {...post}/>)
      }
    </Card.Group>
  )
}

Posts.defaultProps = {
  posts: [],
  fetchControl: {
    status: 'success',
    error: ''
  }
}

export default Posts
