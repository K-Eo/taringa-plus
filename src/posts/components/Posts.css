.post-item {
  display: flex;
  height: 18px;
  margin-bottom: 14px;
  padding-left: 8px;
  width: 100%;
}

.post-item a {
  display: block;
  color: #333;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  width: 100%;
}

.post-item a:visited {
  color: #551A8B;
}

.post-item:last-child {
  margin-bottom: 0;
}

.post-item .glyphicon {
  margin-right: 10px;
}

.post-item:hover {
  color: #333;
}
