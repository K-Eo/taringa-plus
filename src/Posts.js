import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import _ from 'lodash'

import './Posts.css'
import * as actions from './posts/actions'
import { recentSelector, trendingSelector } from './posts/selectors'

const iconsMap = {
  privado: { x: -3, y: 0 },
  juegos: { x: 0, y: -44 },
  imagenes: { x: 0, y: -64 },
  links: { x: 0, y: -86 },
  videos: { x: 0, y: -110 },
  arte: { x: 0, y: -132 },
  offtopic: { x: 0, y: -154 },
  animaciones: { x: 0, y: -174 },
  musica: { x: 0, y: -198 },
  downloads: { x: 0, y: -217 },
  noticias: { x: 0, y: -242 },
  info: { x: 0, y: -243 },
  'tv-peliculas-series': { x: 0, y: -305 },
  patrocinados: { x: 0, y: -332 },
  poringueras: { x: 0, y: -418 },
  gay: { x: 0, y: -507 },
  relatos: { x: 0, y: -528 },
  linux: { x: 0, y: -547 },
  deportes: { x: 0, y: -572 },
  celulares: { x: 0, y: -593 },
  'apuntes-y-monografias': { x: 0, y: -616 },
  comics: { x: 0, y: -637 },
  solidaridad: { x: 0, y: -661 },
  'recetas-y-cocina': { x: 0, y: -678 },
  mac: { x: 0, y: -702 },
  femme: { x: 0, y: -724 },
  'autos-motos': { x: 0, y: -744 },
  humor: { x: 0, y: -769 },
  'ebooks-tutoriales': { x: 0, y: -789 },
  'salud-bienestar': { x: 0, y: -811 },
  taringa: { x: 0, y: -440 },
  'economia-negocios': { x: 0, y: -849 },
  mascotas: { x: 0, y: -866 },
  turismo: { x: 0, y: -890 },
  'manga-anime': { x: 0, y: -912 },
  'ciencia-ficcion': { x: 0, y: -958 },
  'hazlo-tu-mismo': { x: 0, y: -935 },
  ecologia: { x: 0, y: -459 },
}

const Post = (props) =>  {
  const icon = iconsMap[props.category_shortname] || iconsMap['taringa']

  const postStyle = {
    backgroundPosition: `${icon.x}px ${icon.y}px`
  }

  return (
    <li className='post-item'>
      <div className='post-icon' title={props.category_name} style={postStyle}></div>
      <a href={props.canonical} target='_blank'>
        {props.title}
      </a>
    </li>
  )
}

const User = (props) => (
  <a href='/' className='post-item'>
    User {props.i} (1000)
  </a>
)

const Comment = (props) => (
  <div>
    <strong>user</strong>
    &nbsp;
    <a href='/'>
      comment details
    </a>
  </div>
)

class Posts extends Component {
  componentDidMount () {
    this.props.fetchRecent()
    this.props.fetchTrending()
  }
  render () {
    const { posts, trending } = this.props

    return (
      <div>
        <ul className='nav nav-tabs secondary-nav'>
          <li className='active'><Link to='/'>Inicio</Link></li>
          <li className=''><Link to='/'>Novatos</Link></li>
          <li className=''><Link to='/'>Destacados</Link></li>
        </ul>
        <div className='app-container'>
          <div className='row'>
            <div className='col-md-5'>
              <div className='panel panel-default'>
                <div className='panel-heading'>
                  Últimos Posts
                </div>
                <div className='panel-body'>
                  <ul className='list-unstyled'>
                    {
                      posts.map(post => <Post key={post.id} {...post}/>)
                    }
                  </ul>
                </div>
              </div>
            </div>

            <div className='col-md-4'>
              <div className='panel panel-default'>
                <div className='panel-heading'>
                  Posts Destacados
                </div>
                <div className='panel-body'>
                  <ul className='list-unstyled'>
                    {
                      trending.map(post => <Post key={post.id} {...post}/>)
                    }
                  </ul>
                </div>
              </div>

              <div className='panel panel-default'>
                <div className='panel-heading'>
                  Usuarios Top
                </div>
                <div className='panel-body'>
                  {
                    _.times(10, i => (
                      <User key={i} i={i}/>
                    ))
                  }
                </div>
              </div>

              <div className='panel panel-default'>
                <div className='panel-heading'>
                  Últimos Comentarios
                </div>
                <div className='panel-body'>
                  {
                    _.times(10, i => (
                      <Comment key={i} i={i}/>
                    ))
                  }
                </div>
              </div>
            </div>

            <div className='col-md-3'>
              <div className='panel panel-default'>
                <div className='panel-heading'>
                  Lorem ipsum
                </div>
                <div className='panel-body'>
                  Lorem ipsum dolor
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  posts: recentSelector(state),
  trending: trendingSelector(state)
})

export default connect(mapStateToProps, actions)(Posts)
