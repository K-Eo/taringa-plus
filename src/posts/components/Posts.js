import React from 'react'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'

import './Posts.css'
import Panel from './Panel'

import * as actions from '../actions'
import * as selectors from '../selectors'

const Posts = (props) => (
  <div>
    <ul className='nav nav-tabs secondary-nav'>
      <li className='active'><Link to='/'>Inicio</Link></li>
      <li className=''><Link to='/'>Novatos</Link></li>
      <li className=''><Link to='/'>Destacados</Link></li>
    </ul>

    <div className='app-container'>
      <div className='row'>
        <Panel
          col='col-md-5'
          control={props.postsControl}
          items={props.posts}
          load={props.fetchRecent}
          paginator
          refresh={props.fetchRecent}
          maxPages={50}
          title='Ãšltimos Posts'
        />

        <div className='col-md-4'>
          <div className='row'>
            <Panel
              control={props.trendingControl}
              load={props.fetchTrending}
              refresh={props.fetchTrending}
              items={props.trending}
              title='Posts Destacados'
            />
          </div>
          <div className='row'>
            <Panel
              control={props.popularsControl}
              items={props.populars}
              load={props.fetchPopulars}
              refresh={props.fetchPopulars}
              title='Posts Populares'
            />
          </div>
        </div>

        <div className='col-md-3'>
          <div className='panel panel-default'>
            <div className='panel-heading'>
              Lorem ipsum
            </div>
            <div className='panel-body'>
              Lorem ipsum dolor
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
)

const mapStateToProps = (state) => ({
  populars: selectors.popularsSelector(state),
  popularsControl: state.control.populars,
  posts: selectors.recentSelector(state),
  postsControl: state.control.recent,
  trending: selectors.trendingSelector(state),
  trendingControl: state.control.trending,
})

export default connect(mapStateToProps, actions)(Posts)
