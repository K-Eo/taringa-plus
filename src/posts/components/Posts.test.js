import React from 'react'
import { Posts } from './Posts'
import renderer from 'react-test-renderer'

describe('Posts component', () => {
  it('renders posts', () => {
    Date.now = jest.fn(() => 1513924592705)

    const posts = [
      {
        id: 1,
        created: 1513920502005,
        canonical: 'canonical-url-1',
        title: 'post-title-1',
        owner: {
          avatar: {
            medium: 'medium-avatar-1'
          },
          nick: 'owner-nick-1'
        }
      },
      {
        id: 2,
        created: 1513920512005,
        canonical: 'canonical-url-2',
        title: 'post-title-2',
        owner: {
          avatar: {
            medium: 'medium-avatar-2'
          },
          nick: 'owner-nick-2'
        }
      }
    ]

    const component = renderer.create(
      <Posts posts={posts}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders blank slate if not posts', () => {
    const component = renderer.create(
      <Posts/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders loader when fetching posts', () => {
    const control = {
      status: 'fetching',
      error: ''
    }

    const component = renderer.create(
      <Posts fetchControl={control}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders error message when fetching posts fails', () => {
    const control = {
      status: 'failure',
      error: 'network error'
    }

    const component = renderer.create(
      <Posts fetchControl={control}/>
    )

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('calls fetch action', () => {
    const mock = jest.fn()

    const component = renderer.create(
      <Posts fetch={mock}/>
    )

    expect(mock).toBeCalled()
  })
})
