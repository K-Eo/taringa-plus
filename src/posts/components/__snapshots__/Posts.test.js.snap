// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component renders blank slate if not posts 1`] = `
<div
  className="ui icon message"
>
  <i
    aria-hidden="true"
    className="smile icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      No hay posts!
    </div>
    <p>
      Prueba recargando la página para intentar descargar los últimos posts.
    </p>
  </div>
</div>
`;

exports[`Posts component renders error message when fetching posts fails 1`] = `
<div
  className="ui icon negative message"
>
  <i
    aria-hidden="true"
    className="frown icon"
  />
  <div
    className="content"
  >
    <div
      className="header"
    >
      Houston, tenemos un problema!
    </div>
    <p>
      Intenta recargar la página. Error: network error
    </p>
  </div>
</div>
`;

exports[`Posts component renders loader when fetching posts 1`] = `
<div
  className="ui active centered inline loader"
/>
`;

exports[`Posts component renders posts 1`] = `
<div
  className="ui four cards"
>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <img
      className="ui image"
      src=""
    />
    <div
      className="content"
    >
      <a
        className="header"
        href="canonical-url-1"
        target="_blank"
      >
        post-title-1
      </a>
      <div
        className="meta"
      >
        <time
          dateTime="2017-12-22T05:28:22.005Z"
          title="2017-12-22 05:28"
        >
          hace 21 horas
        </time>
      </div>
    </div>
    <div
      className="extra content"
    >
      <img
        className="ui avatar image"
        src="medium-avatar-1"
      />
      owner-nick-1
    </div>
  </div>
  <div
    className="ui card"
    href={undefined}
    onClick={[Function]}
  >
    <img
      className="ui image"
      src=""
    />
    <div
      className="content"
    >
      <a
        className="header"
        href="canonical-url-2"
        target="_blank"
      >
        post-title-2
      </a>
      <div
        className="meta"
      >
        <time
          dateTime="2017-12-22T05:28:32.005Z"
          title="2017-12-22 05:28"
        >
          hace 21 horas
        </time>
      </div>
    </div>
    <div
      className="extra content"
    >
      <img
        className="ui avatar image"
        src="medium-avatar-2"
      />
      owner-nick-2
    </div>
  </div>
</div>
`;
